### PR TITLE
Simpler file chunking

### PR DESCRIPTION
**Pull request by adamjstewart**
Now that Python 3.8 is the minimum supported version, we can use the new [assignment expression](https://docs.python.org/3/whatsnew/3.8.html#assignment-expression) syntax to simplify how we read files chunk-by-chunk.

### CODE REVIEW

**Comment by adamjstewart** [comment]
Ping @ejguan

**Comment by adamjstewart** [comment]
@pmeier can you review since you reviewed https://github.com/pytorch/vision/pull/7673 too?

**Review by pmeier — APPROVED** [review]
Thanks @adamjstewart, LGTM if CI is green. Although I can review, I don't have merge rights here. So we need to wait for the maintainers.

**Comment by adamjstewart** [comment]
Ping ping @ejguan

**Comment by pmeier** [comment]
Sorry to be the bearer of the bad news here, but it seems this PR won't go anywhere in the near future: #1196

**Comment by adamjstewart** [comment]
Thanks for the heads up!

**Pull request closed without merging**
